CONTRIBUTING.md

```diff
@@ -0,0 +1,155 @@
+
+### Moving A Component From `dcos/dcos-ui` To The `ui-kit`
+
+There will be often a chance when you create a component in `DCOS-ui` that it is a good candidate for the ui-kit, but before moving it over to the ui-kit move it to the `candidate` directory in `dcos/dcos-ui` on a regular basis we are going to evaluate usage of components in the `candidate` directory. In that way, we have a good overview of candidate components for the ui-kit.
+
+> Rule of Three:  
+> Try a component in three different places before placing it into ui-kit
+
+### Creating a Reusable Component
+
+#### Avoid Weak Elements
+
+Weak elements are elements which can’t nest block elements, for example, a `p` tag is not able to wrap around a div which will reduce the comparability of a component. This is why we try to rather use a strong element like `section` or `div` instead of `p`.
+
+#### Declare Prop Types
+
+Since the `ui-kit` is using typescript please make sure to define proper prop types for your components. This will improve the documentation, which is created dynamically and will provide better error messages for the users. And also this will provide auto-completion in some IDE’s. Describe the props as good as possible, so also define object shapes or array shapes.
+
+#### Never hard code HTML ID’s
+
+This might conflict at some point. If you really need an id on the element expose a prop for that so the user can define it based on the application.
+
+#### Use Logical Defaults
+
+Use prop defaults which make sense this will reduce the efforts needed to adopt this component.
+
+#### A11y Support
+
+A good start on that topic is to read [this website](https://a11yproject.com/)
+While creating your component think about users: - Which only use keyboards - expose tab indexes as props - Use semantic correct HTML - ARIA
+
+#### Configuration Objects
+
+Please consider using configuration objects as they will provide the user a lot of value by
+
+* consistency
+* less typing needed
+* less error prone
+
+But please avoid using configuration Objects if you only have 3 or fewer properties.
+
+#### Single Responsibility Principle
+
+Keep the components simple by applying them to a single responsibility. In that way, the component will stay simple in most cases. Try not to introduce too many props. Each prop is a configuration which will automatically increase the complexity. Before adding a property to an existing component ask yourself if this could be a new component.
+
+#### Keep Components Dumb
+
+It is important to keep components as dumb as possible, only add logic to a component if it improves performance or quality of a component. For example, paginating a list if the number of items is exceeding a certain amount or having a show toggle to show more content like a dropout.
+
+#### Directory Structure
+
+<!--
+TODO: Decide on directory structure.
+-->
+
+## Atomic Design
```

Comment:
Hey @Poltergeist I don't think we are following atomic design in the UI Kit so I would remove it.

Not just that but after reading it what we have here I don't believe helped me understand how to build a better component.

Reply:
Okay, why do you think we should not follow atomic design in the UI kit? I think it is a very neat pattern to build reusable components. Since in atomic design, you start building atoms which then form molecules and finally organisms. This will provide a high level of consistency. 

I will look into how I can improve the messaging so that we can build better components.

Reply:
My thoughts on this are:

1) I like atomic design a lot, we used it in my last job. It suits design libraries especially well, so IMHO it's a great choice
2) The book is pretty short, so I would like to just point people to it. I think internally we can do maybe some kind of introduction into it, but I agree with @weblancaster that this explanation is not enough
3) I think atomic design falls in natural as the library grows, so it is a good concept to keep in mind and maybe leverage on later when we have 10+ components.